File: mobile-engage/src/main/java/com/emarsys/mobileengage/client/LoggingClientServiceInternal.java
package com.emarsys.mobileengage.client;

import com.emarsys.core.api.result.CompletionListener;
import com.emarsys.core.util.SystemUtils;
import com.emarsys.core.util.log.Logger;
import com.emarsys.core.util.log.entry.MethodNotAllowed;

public class LoggingClientServiceInternal implements ClientServiceInternal {
    private final Class klass;

    public LoggingClientServiceInternal(Class klass) {
        this.klass = klass;
    }

    @Override
    public void trackDeviceInfo(CompletionListener completionListener) {
        String callerMethodName = SystemUtils.getCallerMethodName();

        Logger.debug(new MethodNotAllowed(klass, callerMethodName, null));
    }
}

File: mobile-engage/src/main/java/com/emarsys/mobileengage/client/LoggingClientServiceInternal.kt
package com.emarsys.mobileengage.client

import com.emarsys.core.api.result.CompletionListener
import com.emarsys.core.util.SystemUtils
import com.emarsys.core.util.log.Logger.Companion.debug
import com.emarsys.core.util.log.entry.MethodNotAllowed

class LoggingClientServiceInternal(private val klass: Class<*>) : ClientServiceInternal {
    override fun trackDeviceInfo(completionListener: CompletionListener?) {
        val callerMethodName = SystemUtils.getCallerMethodName()
        debug(MethodNotAllowed(klass, callerMethodName, null))
    }
}